KIALI-1942 grey out showServiceNodes option when graphType=service

jshaughn opened the pull request:
![kiali-1942](https://user-images.githubusercontent.com/2104052/48579040-5d96fb80-e8e9-11e8-9b6d-7dab053bba29.gif)

josejulio commented:
I think it the `Services nodes` checkbox should be `checked` (and disabled) if we are selecting the `graphType=service`, but i think that's for UX to decide.

josejulio commented on src/containers/GraphSettingsContainer.tsx:
```diff
@@ -181,7 +183,7 @@ class GraphSettings extends React.PureComponent<GraphSettingsProps> {
         <label>
-          <input type="checkbox" checked={item.value} onChange={() => item.onChange()} />
+          <input type="checkbox" checked={item.value} onChange={() => item.onChange()} disabled={item.disabled} />
           <span className={checkboxStyle}>{item.labelText}</span>
```
To gray out, you would have to create a new style and apply to this element when the item is disabled:

```tsx
const disabledTextStyle = style({color: 'gray'}); // or PFColors.Gray
// ...
<span className={checkboxStyle item.disabled? disabledTextStyle : ''}>{item.labelText}</span>
```

jshaughn commented:
@josejulio @abonas I've rebased and updated to get the gray text.  Thanks for the tip Josejulio.  As for ensuring the checkbox is checked.  I've thought more about it and recommend we leave it in whatever state it was in when setting the graph type.  I recommend this because if you change the graphType from service to something else the setting will get applied.  And note that in another PR we are changing the default to 'on', so that means if it is off the user manually set it that way, and I don't think we should then toggle it on.

So, this PR can be merged unless you want me to make the toggle change.

josejulio commented:
> @josejulio @abonas I've rebased and updated to get the gray text. Thanks for the tip Josejulio. As for ensuring the checkbox is checked. I've thought more about it and recommend we leave it in whatever state it was in when setting the graph type. I recommend this because if you change the graphType from service to something else the setting will get applied. And note that in another PR we are changing the default to 'on', so that means if it is off the user manually set it that way, and I don't think we should then toggle it on.

I didn't really meant to override the "on/off" setting (e.g. moving the redux value). I was saying that we could have it looking like is "on" when the graphType is service (e.g. is on if setting == 'on' or graphType==service)

That said, I don't know if that could confuse the user.

abonas commented:
> @josejulio @abonas I've rebased and updated to get the gray text. Thanks for the tip Josejulio. As for ensuring the checkbox is checked. I've thought more about it and recommend we leave it in whatever state it was in when setting the graph type. I recommend this because if you change the graphType from service to something else the setting will get applied. And note that in another PR we are changing the default to 'on', so that means if it is off the user manually set it that way, and I don't think we should then toggle it on.
> 
> So, this PR can be merged unless you want me to make the toggle change.

the reasoning here sounds good to me.
Can you just update the most recent screenshot look n feel?

jshaughn commented:
@abonas , I rebased to fix conflicts and have recorded an updated video below.  It shows the greyed out text and also the approach of maintaining the toggle setting.  I'm going to assume this is good and proceed to merge.  I think it's fine but any issue can be addressed in a follow up jira.

![kiali-1942-2](https://user-images.githubusercontent.com/2104052/49297270-63174880-f488-11e8-8e47-6bb30390dea2.gif)
